src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/DynamicInterfaceCastableImplementation.cs

```diff
@@ -0,0 +1,133 @@
+
+        private static readonly DiagnosticDescriptor InterfaceMethodsMissingImplementation =
+            DiagnosticDescriptorHelper.Create(
+                InterfaceMethodsMissingImplementationRuleId,
+                new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.InterfaceMethodsMissingImplementationTitle), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources)),
+                new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.InterfaceMethodsMissingImplementationMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources)),
+                DiagnosticCategory.Usage,
+                RuleLevel.BuildWarning,
+                new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.InterfaceMethodsMissingImplementationDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources)),
+                isPortedFxCopRule: false,
+                isDataflowRule: false);
+
+        internal const string MethodsDeclaredOnImplementationTypeMustBeSealedRuleId = "CA2254";
+
+        private static readonly DiagnosticDescriptor MethodsDeclaredOnImplementationTypeMustBeSealed =
+            DiagnosticDescriptorHelper.Create(
+                MethodsDeclaredOnImplementationTypeMustBeSealedRuleId,
+                new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.MethodsDeclaredOnImplementationTypeMustBeSealedTitle), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources)),
+                new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.MethodsDeclaredOnImplementationTypeMustBeSealedMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources)),
+                DiagnosticCategory.Usage,
+                RuleLevel.BuildWarning,
+                new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.MethodsDeclaredOnImplementationTypeMustBeSealedDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources)),
+                isPortedFxCopRule: false,
+                isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+            DynamicInterfaceCastableImplementationUnsupported,
+            InterfaceMethodsMissingImplementation,
+            MethodsDeclaredOnImplementationTypeMustBeSealed);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+            context.RegisterSymbolAction(context => AnalyzeType(context), SymbolKind.NamedType);
```

Comment:
Should probably work?

```suggestion
            context.RegisterSymbolAction(AnalyzeType, SymbolKind.NamedType);
```

Reply:
It will work, but it ends up actually being slightly worse performance (the delegate instance isn't cached), so I prefer doing the explicit lambda.